beta/src/components/MDX/Sandpack/createFileMap.ts

```diff
@@ -40,14 +50,33 @@ export const createFileMap = (codeSnippets: any) => {
           `File ${filePath} was defined multiple times. Each file snippet should have a unique path name`
         );
       }
+
       result[filePath] = {
         code: (props.children || '') as string,
         hidden: fileHidden,
         active: fileActive,
       };
 
+      if (isTSFile(filePath)) {
+        hasTSVersion = true;
+      }
+
       return result;
     },
     {}
   );
+
+  for (const filePath in fileMap) {
+    // Only remove JS files if we have a TS version available.
+    // If no TS version is available we continue to display JS files.
+    // Assuming that if one file is available as TS, every file is.
+    // Assuming a JS version is available all the time.
+    if (snippetTargetLanguage === 'ts' && isJSFile(filePath) && hasTSVersion) {
+      delete fileMap[filePath];
```

Comment:
Deleting instead of `hidden: true` since hidden files would still be part of the created CodeSandbox.